extensions/tasks-parent/tasks-client/src/main/java/io/resys/thena/tasks/client/spi/visitors/TaskCommandVisitor.java

```diff
@@ -111,28 +137,129 @@ private Task visitCreateTask(CreateTask command) {
     return this.current;
   }
   
-  
   private Task visitChangeTaskStatus(ChangeTaskStatus command) {
     this.current = this.current
         .withStatus(command.getStatus())
-        .withUpdated(command.getTargetDate());
+        .withUpdated(handleTargetDate(command.getTargetDate()));
     return this.current;
   }
   
   private Task visitChangeTaskPriority(ChangeTaskPriority command) {
     this.current = this.current
         .withPriority(command.getPriority())
-        .withUpdated(command.getTargetDate());
+        .withUpdated(handleTargetDate(command.getTargetDate()));
     return this.current;
   }
     
   private Task visitAssignTaskReporter(AssignTaskReporter command) {
     this.current = this.current
         .withReporterId(command.getReporterId())
-        .withUpdated(command.getTargetDate());
+        .withUpdated(handleTargetDate(command.getTargetDate()));
     return this.current;
   }
-  
+
+  private Task visitArchiveTask(ArchiveTask command) {
+    final var targetDate = handleTargetDate(command.getTargetDate());
+    this.current = this.current
+        .withArchived(targetDate)
+        .withUpdated(targetDate);
+    return this.current;
+  }
+
+  private Task visitCommentOnTask(CommentOnTask command) {
+    final var gen = ctx.getGid();
+    final var comments = new ArrayList<>(current.getComments());
+    final var id = Optional.ofNullable(command.getCommentId()).orElseGet(() -> gen.getNextId(DocumentType.TASK));
+    comments.add(ImmutableTaskComment.builder()
+        .id(id)
+        .commentText(command.getCommentText())
+        .replyToId(command.getReplyToCommentId())
+        .username(command.getUserId())
+        .created(handleTargetDate(command.getTargetDate()))
+        .build());
+    this.current = this.current
+        .withComments(comments.stream().toList());
+    return this.current;
+  }
+
+  private Task visitAssignTaskRoles(AssignTaskRoles command) {
+    this.current = this.current
+        .withRoles(command.getRoles().stream().distinct().toList())
```

Comment:
You'll want to sort this list by id-s.  If the order is changed between the role elements, this would appear as a change, even though it's not really.